### ISSUE TITLE
Upgrade appcompat-v7 dependency in cyclestreets-view

### ISSUE THREAD

**Issue by oliverlockwood** [cyclestreets/android#152]
Currently we have the following dependency in `cyclestreets-view`:

``` gradle
  compile 'com.android.support:appcompat-v7:20.+'
```

We should upgrade this to e.g.

``` gradle
  compile 'com.android.support:appcompat-v7:23.4.0'
```

If you try to just make this change, compilation fails with the following stack of errors.  I haven't dug into this further yet.

> /cyclestreets-android/libraries/cyclestreets-view/build/intermediates/res/merged/release/values-ldltr-v21/values-ldltr-v21.xml
> Error:(3) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Spinner.Underlined'.
> /cyclestreets-android/libraries/cyclestreets-view/build/intermediates/res/merged/release/values-v21/values-v21.xml
> Error:(13) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Inverse'.
> Error:(15) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Large.Inverse'.
> Error:(21) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Medium.Inverse'.
> Error:(28) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Small.Inverse'.
> Error:(201) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ProgressBar'.
> Error:(203) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ProgressBar.Horizontal'.
> Error:(211) Error retrieving parent for item: No resource found that matches the given name 'android:Theme.Material'.
> Error:(221) Error retrieving parent for item: No resource found that matches the given name 'android:Theme.Material.Light'.
> Error:(163) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.AutoCompleteTextView'.
> Error:(174) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.EditText'.
> Error:(6) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Button'.
> Error:(3) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material'.
> Error:(4) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Body1'.
> Error:(5) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Body2'.
> Error:(7) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Caption'.
> Error:(8) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Display1'.
> Error:(9) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Display2'.
> Error:(10) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Display3'.
> Error:(11) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Display4'.
> Error:(12) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Headline'.
> Error:(14) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Large'.
> Error:(16) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.PopupMenu.Large'.
> Error:(18) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.PopupMenu.Small'.
> Error:(20) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Medium'.
> Error:(22) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Menu'.
> Error:(23) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.SearchResult.Subtitle'.
> Error:(25) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.SearchResult.Title'.
> Error:(27) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Small'.
> Error:(29) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Subhead'.
> Error:(30) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Title'.
> Error:(31) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Subtitle'.
> Error:(33) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Subtitle.Inverse'.
> Error:(35) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Title'.
> Error:(37) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Title.Inverse'.
> Error:(39) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionMode.Subtitle'.
> Error:(41) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionMode.Title'.
> Error:(43) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button'.
> Error:(44) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.PopupMenu.Large'.
> Error:(46) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.PopupMenu.Small'.
> Error:(48) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Button'.
> Error:(49) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.TextView.SpinnerItem'.
> Error:(50) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Subtitle'.
> Error:(52) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Title'.
> Error:(152) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ActionBar.TabText'.
> Error:(154) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ActionBar.TabView'.
> Error:(156) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ActionButton'.
> Error:(158) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ActionButton.CloseMode'.
> Error:(161) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ActionButton.Overflow'.
> Error:(166) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button'.
> Error:(167) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button.Borderless'.
> Error:(168) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button.Borderless.Colored'.
> Error:(169) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button.Small'.
> Error:(170) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ButtonBar'.
> Error:(171) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.CompoundButton.CheckBox'.
> Error:(172) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.CompoundButton.RadioButton'.
> Error:(173) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.DropDownItem.Spinner'.
> Error:(177) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ImageButton'.
> Error:(178) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Light.ActionBar.TabText'.
> Error:(180) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Light.ActionBar.TabText'.
> Error:(182) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Light.ActionBar.TabView'.
> Error:(184) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Light.PopupMenu'.
> Error:(190) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ListPopupWindow'.
> Error:(192) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ListView'.
> Error:(193) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.ListView.DropDown'.
> Error:(195) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.PopupMenu'.
> Error:(205) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.RatingBar'.
> Error:(206) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.SeekBar'.
> Error:(207) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Spinner'.
> Error:(208) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.TextView.SpinnerItem'.
> Error:(209) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Toolbar.Button.Navigation'.
> /cyclestreets-android/libraries/cyclestreets-view/build/intermediates/exploded-aar/com.android.support/appcompat-v7/23.4.0/res/values-v21/values-v21.xml
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorAccent'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorButtonNormal'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorControlActivated'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorControlHighlight'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorControlNormal'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorPrimary'.
> Error:(58, 5) No resource found that matches the given name: attr 'android:colorPrimaryDark'.
> Error:(102, 5) No resource found that matches the given name: attr 'android:windowElevation'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorAccent'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorButtonNormal'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorControlActivated'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorControlHighlight'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorControlNormal'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorPrimary'.
> Error:(105, 5) No resource found that matches the given name: attr 'android:colorPrimaryDark'.
> Error:(149, 5) No resource found that matches the given name: attr 'android:windowElevation'.
> Error:(186, 5) No resource found that matches the given name: attr 'android:overlapAnchor'.
> Error:(197, 5) No resource found that matches the given name: attr 'android:overlapAnchor'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorAccent'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorButtonNormal'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorControlActivated'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorControlHighlight'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorControlNormal'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorPrimary'.
> Error:(231, 5) No resource found that matches the given name: attr 'android:colorPrimaryDark'.
> /cyclestreets-android/libraries/cyclestreets-view/build/intermediates/res/merged/release/values-v23/values-v23.xml
> Error:(36) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Spinner.Underlined'.
> Error:(3) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.ActionBar.Menu'.
> Error:(4) Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Inverse'.
> Error:(33) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.Button.Colored'.
> Error:(34) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.RatingBar.Indicator'.
> Error:(35) Error retrieving parent for item: No resource found that matches the given name 'android:Widget.Material.RatingBar.Small'.
> Error:Execution failed for task ':libraries:cyclestreets-view:processReleaseResources'.

**Comment by oliverlockwood** [comment]
Need to bump this in priority, because I've run into dependency conflict problems trying to introduce `com.android.support.test.espresso:espresso-contrib` to do more meaningful behavioural testing on the app.

**Comment by jezhiggins** [comment]
It should be fine just to wind this up to v13, so do go ahead.

**Comment by oliverlockwood** [comment]
This has been addressed by #188.  Closing now.
